### PR TITLE
XML description

### PR DESCRIPTION
**Pull request by dcermak**
This is split of #1024 (unfortunately still huge).

TODO:
- [x] drop the first 3 commits once #1069 get's merged and rebase the PR afterwards

Changes proposed in this pull request:
*  Add a new doc folder: `working_with_kiwi` that shall replace `overview`
* Document the RNG schema as it existed for the old KIWI
* Remove `doc/source/building/working_with_images/uninstall_package_requests.rst` as I have incorporated that into `xml_description.rst`

### CODE REVIEW

File: doc/source/working_with_kiwi.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+.. toctree::
+   :maxdepth: 1
+
+   working_with_kiwi/xml_description
+
+
+Overview
+--------
+
+KIWI builds so-called system images (a fully installed and optionally
```
Comment (tomschr) [inline doc/source/working_with_kiwi.rst]: I would introduce new terms in italics:

```suggestion
KIWI builds so-called *system images* (a fully installed and optionally
```

**Comment by dcermak** [comment]
Just FYI: I've removed all unresolved references and replaced them with `FIXME` comments, so this is safe to merge in that regard.

**Comment by schaefi** [comment]
@dcermak could you merge the two commits into one ?

After that this seems ready to be merged

Thanks

**Comment by dcermak** [comment]
Done

File: doc/source/working_with_kiwi/xml_description.rst [hunk 2]
```diff
@@ -0,0 +1,985 @@
...
+
+.. [#f1] `RELAX NG <https://en.wikipedia.org/wiki/RELAX_NG>`_ is a
+         so-called schema language: it describes the structure of a XML
+         document.
+
+.. [#f2] A hybrid persistent filesystem contains a copy-on-write file to
+         keep data persistent over a reboot.
+
+.. FIXME: @schaefi: is this correct?
+.. [#f3] Note that the default primary group is taken from the system on
```
Comment (davidcassany) [inline doc/source/working_with_kiwi/xml_description.rst]: The default primary group should be taken from the system being build. As it is defined in `/etc/login.defs` and `useradd` is called within a chroot, thus I expect useradd to set the default according the built image defaults.

So I think is the other way round :wink:

File: doc/source/working_with_kiwi/xml_description.rst [hunk 3]
```diff
@@ -0,0 +1,985 @@
...
+  will be used
+
+- `components`: Distribution components used for `deb` repositories,
+  defaults to `main`.
+
+- `distribution`: Distribution name information, used for deb repositories.
+
+- `profiles`: List of profiles to which this repository applies.
+
+.. TODO: username & password are supported, but no clue what they are for?
```
[the remark this answers is not in the document]
Comment (dcermak) [reply doc/source/working_with_kiwi/xml_description.rst]: Should I delete them from the RNG schema too?

File: doc/source/working_with_kiwi/xml_description.rst [hunk 4]
```diff
@@ -0,0 +1,985 @@
...
+  of the project `$PROJECT` available on the Open Build Service (OBS). By
+  default KIWI will look for projects on `build.opensuse.org
+  <https://build.opensuse.org>`_, but this can be overridden using the
+  runtime configuration file.
+  Note that it is not possible to add repositories using the `obs://` path
+  from **different** OBS instances (use direct URLs to the :file:`.repo`
+  file instead in this case).
+
+.. FIXME:
+   add this back: .. (see :ref:`The Runtime Configuration File<working-with-kiwi-runtime-configuration-file>`)
```
[the remark this answers is not in the document]
Comment (dcermak) [reply doc/source/working_with_kiwi/xml_description.rst]: in this case yes

File: doc/source/working_with_kiwi/xml_description.rst (hunk 2 above)
Comment (dcermak) [reply doc/source/working_with_kiwi/xml_description.rst]: I'll fix that.

**Pull request merged**